git/filter_protocol.go

```diff
@@ -0,0 +1,114 @@
+package git
+
+import (
+	"bufio"
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"strconv"
+	"strings"
+)
+
+type protocol struct {
```

Comment:
Ah, this should _definitely_ be renamed. I gave this a temporary name when I was working on it last week and never got around to changing it. I'll touch this up in my final PR in the series. 💅